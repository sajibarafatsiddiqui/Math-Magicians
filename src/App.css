.container{
  width:30%;
  padding-top:5rem;
  margin:auto
}

.keyContainer {
  display:grid;
  grid-template-columns: repeat(12,1fr);
  grid-template-rows: repeat(5,1fr);
  grid-gap:0;
  grid-gap:1px
 
}

.darkgray {
  background:#858694;
  text-align:right;
  height:3rem;
  line-height: 3rem;
  color:white;
  font-size:large;
  
}

.lightgray {
  background:#E0E0E0;
  grid-column: auto / span 3;
  text-align: center;
  font-size: small;
  height:3rem;
  line-height: 3rem;
}

.lightgray:nth-child(17){
  grid-column: auto / span 6;
}

.lightgray:nth-child(n+18){
  grid-column: auto / span 3;
}

.lightgray:nth-child(4n){
  background:#F5913E
}

.lightgray:nth-child(19){
  background:#F5913E
}
