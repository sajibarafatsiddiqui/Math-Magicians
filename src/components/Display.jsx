import PropTypes from 'prop-types';

const Display = ({ clname, char }) => <div data-testid="displayCalc" className={clname}>{char}</div>;

Display.propTypes = { clname: PropTypes.string.isRequired, char: PropTypes.string.isRequired };

export default Display;
