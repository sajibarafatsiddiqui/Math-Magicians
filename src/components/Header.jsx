import { NavLink } from 'react-router-dom';

const Header = () => {
  const navlinks = [
    { path: '/math_magicians', title: 'Home' },
    { path: 'Calculator', title: 'Calculator' },
    { path: 'Quote', title: 'Quote' },
  ];

  return (
    <header className="layout">
      <h1>Math Magicians</h1>
      <ul className="nav-links">
        {navlinks.map((navlink) => (
          <li key={navlink.title}>
            <NavLink to={navlink.path}>{navlink.title}</NavLink>
          </li>
        ))}
      </ul>
    </header>
  );
};

export default Header;
