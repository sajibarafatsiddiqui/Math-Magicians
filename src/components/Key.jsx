import PropTypes from 'prop-types';

const Key = ({ clname, char }) => <div className={clname}>{char}</div>;

Key.propTypes = { clname: PropTypes.string.isRequired, char: PropTypes.string.isRequired };

export default Key;
