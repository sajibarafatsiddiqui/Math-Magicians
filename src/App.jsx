import { Route, Routes } from 'react-router-dom';
import Layout from 'router/Layout';
import './App.css';
import Home from 'router/Home';
import CalculatorContainer from 'router/CalculatorContainer';
import Quote from 'router/Quote';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="/math_magicians" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path="Calculator" element={<CalculatorContainer />} />
          <Route path="Quote" element={<Quote />} />
        </Route>
      </Routes>
    </div>
  );
}

export default App;
