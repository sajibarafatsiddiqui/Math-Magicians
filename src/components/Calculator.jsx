import React, { useEffect, useState } from 'react';
import Key from './Key';
import Display from './Display';
import calculate from '../logic/calculate';

const buttons = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
const Calculator = () => {
  const [tools, setTools] = useState({ total: 0, next: 0, operation: null });
  const [display, setDisplay] = useState('0');
  useEffect(() => {
    if (tools.next) {
      setDisplay(tools.next.toString());
    } else if (tools.operation) {
      setDisplay(tools.operation.toString());
    } else if (tools.total) {
      setDisplay(tools.total.toString());
    } else setDisplay('0');
  }, [tools.next, tools.operation, tools.total]);
  const operation = (e) => {
    const updatedTools = calculate(tools, e.target.innerHTML);
    setTools(updatedTools);
  };
  return (
    <section className="container">
      <Display clname="key darkgray" char={display} />
      <div className="keyContainer">{buttons.map((val) => (<Key key={val} clname="key lightgray" char={val} handleClick={operation} />))}</div>
    </section>
  );
};
export default Calculator;
//
// return (
