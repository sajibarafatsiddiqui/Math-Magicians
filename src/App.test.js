import React from 'react';
import operate from 'logic/operate';
import calculate from 'logic/calculate';
import renderer from 'react-test-renderer';
import Quote from 'router/Quote';
import Calculator from 'components/Calculator';
import App from 'App';
import { render } from '@testing-library/react';
import { BrowserRouter, useLocation, NavLink } from 'react-router-dom';
import Header from 'components/Header';
import Key from 'components/Key';
import Display from './components/Display';

describe('components', () => {
  test('renders Quote component correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Calculator component correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Display component correctly', () => {
    const tree = renderer
      .create(<Display clname="key darkgray" char="1" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Header component correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><Header /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Key component correctly', () => {
    const tree = renderer
      .create(<Key />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders App component', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
  });
});

/* Testing Operator.js */

describe('Testing operate.js', () => {
  test('Testing + operator', () => {
    const result = operate(1, 2, '+');
    expect(result).toBe('3');
  });
  test('Testing - operator', () => {
    const result = operate(4, 1, '-');
    expect(result).toBe('3');
  });
  test('Testing x operator', () => {
    const result = operate(0, 3, 'x');
    expect(result).toBe('0');
  });
  test('Testing รท operator', () => {
    const result = operate(15, 2, 'รท');
    expect(result).toBe('7.5');
  });
  test('Testing % operator', () => {
    const result = operate(15, 7, '%');
    expect(result).toBe('1');
  });
});

/* Testing calculate.js */

describe('testing calculate.js ', () => {
  test('Testing AC Button', () => {
    expect(
      calculate(
        {
          total: 5,
          next: 2,
          operation: '+',
        },
        'AC',
      ),
    ).toEqual({
      total: null,
      next: null,
      operation: null,
    });
  });

  test('Testing . Button', () => {
    expect(
      calculate(
        {
          total: null,
          next: '2',
          operation: null,
        },
        '.',
      ),
    ).toEqual({
      total: null,
      next: '2.',
      operation: null,
    });
  });

  test('Testing +/- Button', () => {
    expect(
      calculate(
        {
          total: null,
          next: '2',
          operation: null,
        },
        '+/-',
      ),
    ).toEqual({
      total: null,
      next: '-2',
      operation: null,
    });
  });

  test('Testing +/- Button', () => {
    expect(
      calculate(
        {
          total: null,
          next: '-5',
          operation: null,
        },
        '+/-',
      ),
    ).toEqual({
      total: null,
      next: '5',
      operation: null,
    });
  });
  test('Testing = Button', () => {
    expect(
      calculate(
        {
          total: 5,
          next: '2',
          operation: 'x',
        },
        '=',
      ),
    ).toEqual({
      total: '10',
      next: null,
      operation: null,
    });
  });
});
