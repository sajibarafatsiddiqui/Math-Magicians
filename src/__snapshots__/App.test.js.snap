// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components renders Calculator component correctly 1`] = `
<section
  className="container"
>
  <div
    className="key darkgray"
    data-testid="displayCalc"
  >
    0
  </div>
  <div
    className="keyContainer"
  >
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="key lightgray"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;

exports[`components renders Display component correctly 1`] = `
<div
  className="key darkgray"
  data-testid="displayCalc"
>
  1
</div>
`;

exports[`components renders Header component correctly 1`] = `
<header
  className="layout"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="nav-links"
  >
    <li>
      <a
        className=""
        href="/math_magicians"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</header>
`;

exports[`components renders Key component correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`components renders Quote component correctly 1`] = `
<section
  className="quote"
>
  <h3>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
  </h3>
</section>
`;
